rewrite-core/src/test/java/org/openrewrite/RecipeLifecycleTest.java

```diff
@@ -217,6 +218,114 @@ If the applicability test is incorrectly applied to individual recipes in the li
         );
     }
 
+    @Test
+    void yamlApplicabilityTrueWithRecipesHaveVisitMethodOverridden() {
+        //language=yaml
+        String yamlRecipe = """
+          ---
+          type: specs.openrewrite.org/v1beta/recipe
+          name: org.openrewrite.RecipesToTransformMultiFiles
+          displayName: Recipes in the list which override 'visit()' method should be run if applicability test pass
+          description: >
+            A recipe has two different ways to run.
+            1. Override 'getVisitor()' method, and invoke `getVisitor().visit(@Nullable Tree tree, P p)` to transform a single file.
+            2. Override `visit(List<SourceFile> before, ExecutionContext ctx)` method, and invoke it to transform multiple files.
+            Typically, for a recipe, only one of the two methods mentioned above is required to be overridden.
+            The recipe scheduler invokes both methods in different places in the flow, this test is intended to make sure 
+            those recipes that overrides 'visit()' method can be run correctly with the applicability test.
+            
+            All recipes in the list should be run only when the applicability test pass.
+            
+            Given a text file containing the number "1", running this recipe should result in a file which has text "1->2->3".
+          applicability:
+            singleSource:
+              - org.openrewrite.text.FindAndReplace:
+                  find: "1"
+                  replace: "A"
+          recipeList:
+            - org.openrewrite.text.AppendToTextFile:
+                  relativeFileName: "file.txt"
+                  content: "->2"
+                  preamble: "preamble"
+                  appendNewline : false
+                  existingFileStrategy: "continue"
+            - org.openrewrite.text.AppendToTextFile:
+                  relativeFileName: "file.txt"
+                  content: "->3"
+                  preamble: "preamble"
+                  appendNewline : false
+                  existingFileStrategy: "continue"
+          """;
+        rewriteRun(
+          spec -> spec.recipe(Environment.builder()
+            .scanRuntimeClasspath()
+            .load(
+              new YamlResourceLoader(
+                new ByteArrayInputStream(yamlRecipe.getBytes(StandardCharsets.UTF_8)),
+                Paths.get("applicability.yml").toUri(),
+                new Properties()))
+            .build()
+            .activateRecipes("org.openrewrite.RecipesToTransformMultiFiles")),
+          text("1",
+            "1->2->3",
+            spec -> spec.path("file.txt").noTrim())
+        );
+    }
+
+    @Disabled
+    @Test
+    void yamlApplicabilityFalseWithRecipesHaveVisitMethodOverridden() {
```

Comment:
If we already have an issue number for this add `@Issue("https://....")`

Reply:
Oh, Yeah, I created an issue ticket after this PR, let me add the @Issue link here.

Reply:
added.